### PR TITLE
Better Handling of Address Parsing Errors

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

The `InterfaceInfo.CIDRAddress` method previously returned any raised errors as the result of the method. This means callers could receive a stringified error without any indication that it was not actually a parsed address.

This has probably not been being encountered in the wild, because we would have seen such error messages rendered in the `ip.addresses` Mongo collection and/or rendered as invalid netplan/ENI config.

This patch adds an error return to the method, which is handled at the various call sites.

## QA steps

Execute the following and check logs for errors.

- Bootstrap to LXD (uses the method when rendering network config).
  - Add a machine.
- Bootstrap to OpenStack (uses the method when rendering network config).
  - Add a machine and then a container machine (container broker uses the method).

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Better Error Handling for Address Parsing](https://trello.com/c/V6aEBQGP/2592-better-error-handling-for-address-parsing)

File: core/network/nic.go [hunk 1]
```diff
@@ -211,21 +211,21 @@ func (i *InterfaceInfo) IsVLAN() bool {
 }
 
 // CIDRAddress returns Address.Value combined with CIDR mask.
-func (i *InterfaceInfo) CIDRAddress() string {
+func (i *InterfaceInfo) CIDRAddress() (string, error) {
 	primaryAddr := i.PrimaryAddress()
 	if i.CIDR == "" || primaryAddr.Value == "" {
-		return ""
+		return "", nil
```
Comment (achilleasa) [inline core/network/nic.go:217]: Why not return an `errors.NotValid` here? It feels strange to have a method returning an empty value when it encounters errors and an `"", nil` here.

Comment (manadart) [reply core/network/nic.go]: I returned a `NotFound`, which I think is more appropriate given the usage context for cloud-init generation. In this case, we fall back to DHCP for the device.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**